### ISSUE TITLE
Need to find out the reason for the following error

### ISSUE THREAD

**Issue by houshengbo** [knative/operator#652]
When I enable all the available eventing source, I got the message in the log:
```
E0608 02:13:11.862468       1 leaderelection.go:361] Failed to update lock: Put "https://10.96.0.1:443/apis/coordination.k8s.io/v1/namespaces/default/leases/knative-operator.knative.dev.operator.pkg.reconciler.knativeeventing.reconciler.00-of-01": context deadline exceeded
I0608 02:13:11.906384       1 leaderelection.go:278] failed to renew lease default/knative-operator.knative.dev.operator.pkg.reconciler.knativeeventing.reconciler.00-of-01: timed out waiting for the condition
E0608 02:13:45.077901       1 leaderelection.go:325] error retrieving resource lock default/knative-operator.knative.dev.operator.pkg.reconciler.knativeserving.reconciler.00-of-01: Get "https://10.96.0.1:443/apis/coordination.k8s.io/v1/namespaces/default/leases/knative-operator.knative.dev.operator.pkg.reconciler.knativeserving.reconciler.00-of-01": context deadline exceeded
I0608 02:13:45.379732       1 leaderelection.go:278] failed to renew lease default/knative-operator.knative.dev.operator.pkg.reconciler.knativeserving.reconciler.00-of-01: timed out waiting for the condition
E0608 02:13:46.499747       1 leaderelection.go:301] Failed to release lock: resource name may not be empty
{"severity":"INFO","timestamp":"2021-06-08T02:13:46.5975328Z","logger":"knative-operator","caller":"leaderelection/context.go:156","message":"\"knative-operator-6cfb5cd68c-7f7cs_e6cfad38-c81c-4a53-b82b-c6b558dcfd18\" has stopped leading \"knative-operator.knative.dev.operator.pkg.reconciler.knativeserving.reconciler.00-of-01\"","knative.dev/pod":"knative-operator-6cfb5cd68c-7f7cs"}
I0608 02:13:46.685743       1 leaderelection.go:243] attempting to acquire leader lease  default/knative-operator.knative.dev.operator.pkg.reconciler.knativeserving.reconciler.00-of-01...
E0608 02:13:47.306247       1 leaderelection.go:301] Failed to release lock: rpc error: code = Unavailable desc = transport is closing
{"severity":"INFO","timestamp":"2021-06-08T02:13:47.3136023Z","logger":"knative-operator","caller":"leaderelection/context.go:156","message":"\"knative-operator-6cfb5cd68c-7f7cs_fb0cf69e-6fa2-41fc-be58-6810c9b8edae\" has stopped leading \"knative-operator.knative.dev.operator.pkg.reconciler.knativeeventing.reconciler.00-of-01\"","knative.dev/pod":"knative-operator-6cfb5cd68c-7f7cs"}
I0608 02:13:47.542685       1 leaderelection.go:243] attempting to acquire leader lease  default/knative-operator.knative.dev.operator.pkg.reconciler.knativeeventing.reconciler.00-of-01...
I0608 02:14:07.286395       1 leaderelection.go:253] successfully acquired lease default/knative-operator.knative.dev.operator.pkg.reconciler.knativeeventing.reconciler.00-of-01
{"severity":"INFO","timestamp":"2021-06-08T02:14:07.3245182Z","logger":"knative-operator","caller":"leaderelection/context.go:147","message":"\"knative-operator-6cfb5cd68c-7f7cs_fb0cf69e-6fa2-41fc-be58-6810c9b8edae\" has started leading \"knative-operator.knative.dev.operator.pkg.reconciler.knativeeventing.reconciler.00-of-01\"","knative.dev/pod":"knative-operator-6cfb5cd68c-7f7cs"}
I0608 02:14:07.386920       1 leaderelection.go:253] successfully acquired lease default/knative-operator.knative.dev.operator.pkg.reconciler.knativeserving.reconciler.00-of-01
```

Many webhook deployments went down:
```
NAME                          READY   UP-TO-DATE   AVAILABLE   AGE
ceph-controller               1/1     1            1           6m16s
ceph-webhook                  0/1     1            0           6m15s
couchdb-webhook               0/1     1            0           6m11s
eventing-controller           1/1     1            1           6m29s
eventing-webhook              0/1     1            0           6m28s
github-webhook                0/1     1            0           6m7s
gitlab-controller-manager     1/1     1            1           6m3s
gitlab-webhook                1/1     1            1           5m59s
imc-controller                0/1     1            0           6m24s
imc-dispatcher                1/1     1            1           6m23s
mt-broker-controller          1/1     1            1           6m19s
mt-broker-filter              1/1     1            1           6m21s
mt-broker-ingress             1/1     1            1           6m20s
natss-ch-controller           1/1     1            1           5m54s
natss-ch-dispatcher           1/1     1            1           5m53s
natss-webhook                 0/1     1            0           5m38s
pingsource-mt-adapter         0/0     0            0           6m29s
prometheus-source-webhook     1/1     1            1           5m16s
rabbitmq-controller-manager   1/1     1            1           4m53s
rabbitmq-webhook              0/1     1            0           4m44s
sugar-controller              1/1     1            1           6m18s
```

**Comment by evankanderson** [comment]
This looks like the apiserver might be overloaded; have you looked at the cpu usage of the apiserver during this time?

Any cluster details available to share? Minikube? Kind? Cloud provider? How many nodes, and how much cpu and ram?

`kubectl describe nodes` will have some of this information, but not the apiserver cpu usage, which you many be able to get using `top` on the node or in the apiserver container.

**Comment by houshengbo** [comment]
I am using a single node of kubernetes at docker-desktop. 2G memory and 4 CPUs

**Comment by houshengbo** [comment]
This issue miraculously went away after I fixed this issue in couchdb eventing source: https://github.com/knative-sandbox/eventing-couchdb/issues/112.

On top of deocker-desktop, I cannot produce the couchdb source issue, but on top of IKS, the deployment couchdb-webhook is unbale to reach the running status due to the RBAC issue.

After I fixed the RBAC issue with couchdb source, this issue with the operator so far has not shown up after running operator with eventing sources for almost an hour. No kube scheduler crashing. If so, it could be a great surprise in my favor. I will certainly accept this bonus.
